src/Playwright.MSTest/PlaywrightTestMethodAttribute.cs

```diff
@@ -0,0 +1,67 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Playwright.TestAdapter;
+using MSTestUnitTesting = Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Playwright.MSTest
+{
+    public class PlaywrightTestMethodAttribute : MSTestUnitTesting.TestMethodAttribute
+    {
+        // First run is always 0, for each subsequent run, the run count is incremented by 1.
+        internal static readonly IDictionary<string, int> Test2RetryCount = new Dictionary<string, int>();
+
+        public override MSTestUnitTesting.TestResult[] Execute(MSTestUnitTesting.ITestMethod testMethod)
+        {
+            var key = testMethod.TestClassName + "." + testMethod.TestMethodName;
+            MSTestUnitTesting.TestResult[] results = new MSTestUnitTesting.TestResult[] { };
+            Test2RetryCount[key] = -1;
+            while (Test2RetryCount[key] < PlaywrightSettingsProvider.Retries)
+            {
+                Test2RetryCount[key]++;
+                results = base.Execute(testMethod);
```

Comment:
Shouldn't we append to results here, so that line 64 returns all results?

Reply:
After a discussion with pf we wanted to implemented for now a primitive approach, since if we return a failed test inside that list, it will result with exitCode 1 which we want to prevent.